### PR TITLE
[To Do] fix constructor issues in todo skills

### PR DESCRIPTION
**Pull request by feich-ms**
Fix constructor issues in todo skills
## Description
1. Move all the services in todo skill to service manager
2. Only service manager is kept in the constructor
3. Init all the services by service manager
3. Fix tests accordingly

## Related Issue
#491

## Testing Steps
Launch VA to test to do skill

## Checklist
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the appropriate unit tests

### CODE REVIEW

File: solutions/Virtual-Assistant/src/csharp/skills/todoskill/Dialogs/Main/MainDialog.cs [hunk 1]
```diff
@@ -335,6 +333,18 @@ private void InitializeConfig(ToDoSkillState state)
 
                 state.ReadSize = readSize <= 0 || readSize > CommonUtil.MaxReadSize ? CommonUtil.MaxReadSize : readSize;
             }
+
+            if (state.TaskServiceType == ProviderTypes.Other)
+            {
+                state.TaskServiceType = ProviderTypes.Outlook;
+                if (_services.Properties.ContainsKey("TaskServiceProvider"))
```
Comment (lzc850612) [inline solutions/Virtual-Assistant/src/csharp/skills/todoskill/Dialogs/Main/MainDialog.cs]: use _services.Properties.TryGetValue("TaskServiceProvider")

**Review by lzc850612 — CHANGES_REQUESTED** [review]

File: solutions/Virtual-Assistant/src/csharp/skills/todoskill/Dialogs/Main/MainDialog.cs (hunk 1 above)
Comment (feich-ms) [reply solutions/Virtual-Assistant/src/csharp/skills/todoskill/Dialogs/Main/MainDialog.cs]: fixed

File: solutions/Virtual-Assistant/src/csharp/skills/todoskill/Dialogs/Main/MainDialog.cs [hunk 2]
```diff
@@ -305,21 +303,21 @@ private async Task<InterruptionAction> OnLogout(DialogContext dc)
...
-            // Initialize PageSize and ReadSize when the first input comes.
+            // Initialize PageSize, ReadSize and TaskServiceType when the first input comes.
             if (state.PageSize <= 0)
             {
                 int pageSize = 0;
-                if (_services.Properties.ContainsKey("DisplaySize"))
+                if (_services.Properties.TryGetValue("DisplaySize", out object displaySizeObj))
                 {
-                    pageSize = int.Parse(_services.Properties["DisplaySize"].ToString());
+                    int.TryParse(displaySizeObj.ToString(), out pageSize);
```
Comment (lzc850612) [inline solutions/Virtual-Assistant/src/csharp/skills/todoskill/Dialogs/Main/MainDialog.cs:320]: you need to handle the case where the int.TryParse fails. Log that exception and show some message to user saying its a bot issue.

**Review by lzc850612 — CHANGES_REQUESTED** [review]

File: solutions/Virtual-Assistant/src/csharp/skills/todoskill/Dialogs/Main/MainDialog.cs (hunk 2 above)
Comment (feich-ms) [reply solutions/Virtual-Assistant/src/csharp/skills/todoskill/Dialogs/Main/MainDialog.cs:320]: @lzc850612 Here we have default value for "PageSize"(CommonUtil.MaxDisplaySize) and "ReadSize"(CommonUtil.MaxReadSize). This is the reason why I don't want to throw exception here. The flow should not be blocked by this minor config issue. If developers see the unexpected behavior of their config setting not working, they should be able to debug and find the cause. So I prefer not to throw exception here. Please let me know if you think the exception must be thrown here. Thanks.

Comment (lzc850612) [reply solutions/Virtual-Assistant/src/csharp/skills/todoskill/Dialogs/Main/MainDialog.cs:320]: 'PageSize' is a hard dependency on the appsetting. what if this appsetting is compromised or have an illegal value? Do we just throw an unhandled exception or do we show something useful to the user and log the exception so the developer can look at it? We need to fail gracefully for all cases.

**Review by lzc850612 — APPROVED** [review]

**Pull request merged**